lib/src/address_form.dart

```diff
@@ -0,0 +1,192 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+import 'package:flutter_address_form/src/models/address_model.dart';
+
+/// A widget that creates a form with different address fields widgets.
+/// Returns a `AddressModel` Object from a `AddressController`.
+class AddressForm extends StatelessWidget {
+  AddressForm({
+    Key? key,
+    this.zipCodeLabel = const Text('Zipcode'),
+    this.housenumberLabel = const Text('Housenumber'),
+    this.suffixLabel = const Text('Suffix'),
+    this.streetLabel = const Text('Street'),
+    this.cityLabel = const Text('City'),
+    required this.zipCodeValidator,
+    required this.housenumberValidator,
+    required this.suffixValidator,
+    required this.streetValidator,
+    required this.cityValidator,
+    AddressController? controller,
+  }) {
+    _addressController =
+        controller ?? AddressController(onAutoComplete: (model) => model);
+  }
+
+  final Widget zipCodeLabel;
+  final Widget housenumberLabel;
+  final Widget suffixLabel;
+  final Widget streetLabel;
+  final Widget cityLabel;
+
+  final String? Function(String) zipCodeValidator;
+  final String? Function(String) housenumberValidator;
+  final String? Function(String) suffixValidator;
+  final String? Function(String) streetValidator;
+  final String? Function(String) cityValidator;
+
+  /// Controls the `AddressModel`
+  late final AddressController _addressController;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        AddressFormTextField(
+          controller: _addressController._zipcodeController,
+          validator: zipCodeValidator,
+          label: zipCodeLabel,
+        ),
+        Flexible(
+          child: Row(
+            children: [
+              AddressFormTextField(
+                controller: _addressController._housenumberController,
+                validator: housenumberValidator,
+                label: housenumberLabel,
+              ),
+              AddressFormTextField(
+                controller: _addressController._suffixController,
+                validator: suffixValidator,
+                label: suffixLabel,
+              ),
+            ],
+          ),
+        ),
+        AddressFormTextField(
+          controller: _addressController._streetController,
+          validator: streetValidator,
+          label: streetLabel,
+        ),
+        AddressFormTextField(
+          controller: _addressController._cityController,
+          validator: cityValidator,
+          label: cityLabel,
+        ),
+      ],
+    );
+  }
+}
+
+class AddressFormTextField extends StatelessWidget {
+  final Widget label;
+  final TextEditingController controller;
+  final String? Function(String) validator;
+  const AddressFormTextField({
+    Key? key,
+    required this.label,
+    required this.controller,
+    required this.validator,
+  }) : super(key: key);
+
+  String? get _errorText => validator(controller.value.text);
```

Comment:
the validator should only trigger when someone tries to submit, so maybe try to add a AddressFormController.validate() that makes all validators run and returns a bool wether it is validated